src/main/java/net/reini/rabbitmq/cdi/ConnectionManagerWatcherThread.java

```diff
@@ -43,7 +43,7 @@ boolean isRunning() {
 
   private void ensureConnectionState() {
 
-    while (!interrupted()) {
+    while (!Thread.currentThread().isInterrupted() && stopped == false) {
```

Comment:
you can also use  the convenience method `Thread.interrupted()` in favor of `Thread.currentThread().isInterrupted()`